### ISSUE TITLE
Feature/Improvement Request: Guide Types in the Crop Module

### ISSUE THREAD

**Issue by EOBeav** [Beep6581/RawTherapee#6085]
I'm a new GitHub user so hopefully I'm addressing this request properly.

RawTherapee needs improved guide types in the Crop module. Ideally, the user could create and upload their own guides for this purpose. However, a simpler and more immediate solution might be just to improve and expand upon the guides that are hard coded into the system. I'm not a coder, but if it's a matter of just creating additional .png files for the guides, I'd be more than happy to contribute. 

Specifically, the addition of a Fibonacci ratio and golden triangle would be helpful, as would eliminating the existing guides that seem to be redundant.

**Comment by TechXavAL** [comment]
Isn't the Fibonacci ratio the same as the Harmonic means?

And the Golden triangle is already implemented in 2 versions, isn't it?

Custom guides would be nice, by the way.

**Comment by ghost** [comment]
`Isn't the Fibonacci ratio the same as the Harmonic means?`
Not in this context.

Harmonic means / Golden sections:
![hm](https://user-images.githubusercontent.com/56407344/106393545-9864ab80-63ef-11eb-8f3c-d953197646fb.png)

Fibonacci ratio / Golden spiral:
![Fibonacci_spiral](https://user-images.githubusercontent.com/56407344/106393407-d57c6e00-63ee-11eb-9597-0f3b34eba9c0.gif)

These fit perfectly together though.

`And the Golden triangle is already implemented in 2 versions, isn't it?`
Yes they are (Golden Triangles 1 and 2).

There are other ways of showing them though. Not sure what @EOBeav has in mind exactly.

**Comment by EOBeav** [comment]
>  Yes they are (Golden Triangles 1 and 2).

Yes, there are a few different ways of displaying the golden triangle (or just triangles in general, golden or otherwise).  I guess in this case custom guides would be ideal. The Fibonacci ratio guide above would be important, as well as the different combinations of overlay.

**Comment by ff2000** [comment]
I often miss plain simple center lines. I do not care too much about positioning perfectly via harmonic means/rule of thirds/... because usually subjects are big and "chaotic" enough to make it hard to pick the center. But I like to judge if right/left halves are balanced. Or I want to align the subject perfectly horicontally centered.
I also tend to abuse the grid lines for leveling. I often struggle to find a way to pull the leveling line, having a visual clue makes it easier for me.

How would you like an interactive tool? One vertical, one horizontal line that can be placed freely, some marks for rule of thirds/harmonic means/... on the side.

**Comment by ghost** [comment]
@ff2000 

> How would you like an interactive tool? One vertical, one horizontal line that can be placed freely, some marks for rule of thirds/harmonic means/... on the side.

That would certainly be a welcome expansion, especially if you would add 2 diagonals as well.

BTW: I don't think all this needs to necessarily be integrated into the current crop tool, it can be a stand alone addition. The way the current crop tool works is that if you set, say, rule of thirds it stays visible even if you start working with another module. Same principle could be applied to a possible new *Guides and Perspectives* module.

**Comment by Thanatomanic** [comment]
There is a long-standing request for guides independent of the crop tool, see here https://github.com/Beep6581/RawTherapee/issues/504
I am very much in favour of this idea, but who is going to do it? 😄 

> Specifically, the addition of a Fibonacci ratio and golden triangle would be helpful, as would eliminating the existing guides that seem to be redundant.

@EOBeav Can you elaborate further on which changes you would actually want? The request in your initial post is not overly specific.


The idea of custom guides is interesting, but possibly quite hard to do. Maybe it would be possible to overlay a scalable vector SVG on top of the preview window. My gut tells me this is also a can of worms. Is there any other photographic application that has the ability to set custom guides? In desktop publishing I guess setting guides for aligning stuff is second nature.

**Comment by ghost** [comment]
@EOBeav  and I both created the same issue within minutes of each other. I asked them to be merged, which I now understand cannot be done. I'll copy/paste my FR here, as suggested by @Thanatomanic when he closed mine, so there's an hopefully better all-round idea what it is that "we" want. 

---
*copy/past starts here:*

The request: The possibility for the user to create and add guides to the crop module.

This was asked for on pixls ([Additional frame guides in crop tool](https://discuss.pixls.us/t/additional-frame-guides-in-crop-tool/23145/)).

@heckflosse mentioned that the current guides are hardcoded. I'm not a developer and thus have no idea how hard it is to implement a 'dynamic guide' possibility,  so adding a few hardcoded extra guides would be a second choice but very welcome nonetheless!

The 9 guides available to the  user at the moment do not include some types and/or variations that might be beneficial to the user. The 2 that jump out, for me that is, are:

- Additions to the current Harmonic means guide: A stand alone  spiral (fluent and sectioned) and the spiral on top of the current harmonic means. You do need to be able to flip these horizontally and/or vertically.
- A simple circular perspective guide.

The reason I would like to see a solution that makes it possible for the user to add guides has to do with the almost limitless amount of guides that can be produced and used. I have been looking into dynamic symmetry lately and adding the "standard" ones alone would make the list way to long for normal use.
